node/libs/protobuf_build/src/lib.rs

```diff
@@ -0,0 +1,297 @@
+    pub fn load(&self, pool: &mut prost_reflect::DescriptorPool) -> anyhow::Result<()> {
+        if self
+            .descriptor_proto
+            .file
+            .iter()
+            .all(|f| pool.get_file_by_name(f.name()).is_some())
+        {
+            return Ok(());
+        }
+        for d in &self.dependencies {
+            d.load(pool)?;
+        }
+        pool.add_file_descriptor_set(self.descriptor_proto.clone())?;
+        Ok(())
+    }
+
+    /// Loads the descriptor to the global pool and returns a copy of the global pool.
+    pub fn get_message_by_name(&self, name: &str) -> Option<prost_reflect::MessageDescriptor> {
+        /// Global descriptor pool.
+        static POOL: Lazy<Mutex<prost_reflect::DescriptorPool>> = Lazy::new(Mutex::default);
+        let pool = &mut POOL.lock().unwrap();
+        self.load(pool).unwrap();
+        pool.get_message_by_name(name)
+    }
+}
+
+/// Expands to a descriptor declaration.
+#[macro_export]
+macro_rules! declare_descriptor {
+    ($package_root:expr, $descriptor_path:expr, $($rust_deps:path),*) => {
+        pub static DESCRIPTOR : $crate::Lazy<$crate::Descriptor> = $crate::Lazy::new(|| {
+            $crate::Descriptor::new(
+                $package_root.into(),
+                vec![$({ use $rust_deps as dep; &dep::DESCRIPTOR }),*],
+                &include_bytes!($descriptor_path)[..],
+            )
+        });
+    }
+}
+
+/// Code generation config. Use it in build scripts.
+pub struct Config {
+    /// Input directory relative to $CARGO_MANIFEST_DIR with the proto files to be compiled.
+    pub input_root: InputPath,
+    /// Implicit prefix that should be prepended to proto paths of the proto files in the input directory.
+    pub proto_root: ProtoPath,
+    /// Descriptors of the dependencies and the rust absolute paths under which they will be available from the generated code.
+    pub dependencies: Vec<(RustName, &'static Descriptor)>,
+    /// Rust absolute path under which the protobuf crate will be available from the generated
+    /// code.
+    pub protobuf_crate: RustName,
+}
+
+impl Config {
+    /// Location of the protobuf_build crate, visible from the generated code.
+    fn this_crate(&self) -> RustName {
+        self.protobuf_crate.clone().join("build")
+    }
+
+    /// Generates implementation of `prost_reflect::ReflectMessage` for a rust type generated
+    /// from a message of the given `proto_name`.
+    fn reflect_impl(&self, proto_name: &ProtoName) -> anyhow::Result<String> {
+        let rust_name = proto_name
+            .relative_to(&self.proto_root.to_name().context("invalid proto_root")?)
+            .unwrap()
+            .to_rust_type()
+            .to_string();
+        let rust_name: syn::Path = syn::parse_str(&rust_name).context("rust_name")?;
+        let proto_name = proto_name.to_string();
+        let this: syn::Path = syn::parse_str(&self.this_crate().to_string()).context("this")?;
```

Comment:
Nit: A more conventional way to do this is to use [`parse_quote!`](https://docs.rs/syn/latest/syn/macro.parse_quote.html), or make `RustName` convertible to `syn::Path` (IIUC, `RustName` *is* semantically equivalent to it). With the current `RustName` internal structure, this is hardly possible.